### PR TITLE
Update RequireResource to handle duplicate short names and proper fully qualified names (also make life a little better for windows file system contributors)

### LINKED ISSUE #7070: Applications.Core/secretStores list and delete cli commands giving wrong data.
### Steps to reproduce

Deploy a bicep with secret store resource
```
 ~/radius> rad deploy corerp-resources-gateway-tlstermination.bicep

Building test/functional/shared/resources/testdata/corerp-resources-gateway-tlstermination.bicep...
Deploying template 'test/functional/shared/resources/testdata/corerp-resources-gateway-tlstermination.bicep' into environment 'default' from workspace 'dev'...

Deployment In Progress...

Completed            corerp-resources-gateway-tlstermination Applications.Core/applications
..                   tls-gtwy-front-ctnr Applications.Core/containers
Completed            tls-gtwy-cert1312 Applications.Core/secretStores
..                   tls-gtwy-gtwy   Applications.Core/gateways

Deployment Complete

Resources:
    corerp-resources-gateway-tlstermination Applications.Core/applications
    tls-gtwy-front-ctnr Applications.Core/containers
    tls-gtwy-gtwy   Applications.Core/gateways
    tls-gtwy-cert1312 Applications.Core/secretStores

Public Endpoints:
    tls-gtwy-gtwy   Applications.Core/gateways https://tls-gtwy-gtwy.corerp-resources-gateway-tlstermination.172.19.0.3.nip.io
 ~/radius > rad resource list secretstores
```
List the secrets deployed in the above bicep file
```
 ~/Documents/RadiusProject/radius > rad resource list secretstores                                                                                        
RESOURCE  TYPE      GROUP     STATE
```
Returns empty list and same with delete command.

### Observed behavior

With the split namespace changes we updated `Applications.Link/daprSecretStores` -> `Applications.Dapr/secretStores` and   in `rad resource` commands we only take the resource type(i.e `secretStores` and not `Applications.Dapr/secretStores`) as input and  it matches with the resource type of `Application.Core/secretStores`
image

Because of the same name for resource-type it either works for daprSecrets/coreSecrets. Since Applications.Dapr/secretStores is above Applications.Dapr/secretStores here in this list, list and delete commands are looking for daprSecrets.
https://github.com/radius-project/radius/blob/479bd51f9a92bf5c1ab98b3e83dffdd4cb1a4568/pkg/cli/clients/management.go#L52-L67

### Desired behavior

resource list and delete commands for Application.Core/secretStores should  provide right data i.e information on the coreSecrets rather than daprSecrets.

### Workaround

https://github.com/radius-project/radius/blob/479bd51f9a92bf5c1ab98b3e83dffdd4cb1a4568/pkg/cli/clients/management.go#L52-L67

moved `sstr_ctrl.ResourceTypeName` to top of the list to get unblocked.

Long term fix could be:

Option1 : Provide complete resource type in rad resource commands for all resources.
e.g:

`rad resource list Application.Core/secretStores`  (or)  `rad resource list Application.Dapr/secretStores`

Option2 : change the type of `Application.Dapr/secretStores` from secretStores to daprSecretStores in cli argument.

`rad resource list daprSecretStores`

### rad Version

```
$ rad version
RELEASE   VERSION   BICEP     COMMIT
0.29.0    v0.29.0   0.29.0    6abd7bfc3de0e748a2c34b721d95097afb6a2bba
```

### Operating system

macOS, Sanoma, M1 chip

### Additional context

_No response_

### Would you like to support us?

- [ ] Yes, I would like to support you

[AB#10984](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/10984)

### LINKED ISSUE #7121: Tests fail on Windows when Repo setup in Windows filesystem before starting Devcontainer.
### Steps to reproduce

Pull source code from GH client in Windows cmd (or GH desktop). 
Start Devcontainer using VS Code Devcontainer extension
Run make build
Run make test

tests that rely on comparing sample files (like terraform recipe tests) fail due to test files being checked out with LFCR (\r\n) but tests expecting only LF (\n)

### Observed behavior
Example results from terraform test.
```
expected  : &terraform.moduleInspectResult{ContextVarExists:true, RequiredProviders:[]string{"aws"}, ResultOutputExists:true, Parameters:map[string]interface {}{"context":map[string]interface {}{"defaultValue":interface {}(nil), "description":"This variable contains Radius recipe context.", "name":"context", "pos":tfconfig.SourcePos{Filename:"testdata/.terraform/modules/test-module-recipe-context-outputs/variables.tf", Line:1}, "required":true, "sensitive":false, "type":"object({\n    resource = object({\n      name = string\n      id = string\n      type = string\n    })\n\n    application = object({\n      name = string\n      id = string\n    })\n\n    environment = object({\n      name = string\n      id = string\n    })\n\n    runtime = object({\n      kubernetes = optional(object({\n        namespace = string\n        environmentNamespace = string\n      }))\n    })\n\n    azure = optional(object({\n      resourceGroup = object({\n        name = string\n        id = string\n      })\n      subscription = object({\n        subscriptionId = string\n        id = string\n      })\n    }))\n    \n    aws = optional(object({\n      region = string\n      account = string\n    }))\n  })"}}}
actual  : &terraform.moduleInspectResult{ContextVarExists:true, RequiredProviders:[]string{"aws"}, ResultOutputExists:true, Parameters:map[string]interface {}{"context":map[string]interface {}{"defaultValue":interface {}(nil), "description":"This variable contains Radius recipe context.", "name":"context", "pos":tfconfig.SourcePos{Filename:"testdata/.terraform/modules/test-module-recipe-context-outputs/variables.tf", Line:1}, "required":true, "sensitive":false, "type":"object({\r\n    resource = object({\r\n      name = string\r\n      id = string\r\n      type = string\r\n    })\r\n\r\n    application = object({\r\n      name = string\r\n      id = string\r\n    })\r\n\r\n    environment = object({\r\n      name = string\r\n      id = string\r\n    })\r\n\r\n    runtime = object({\r\n      kubernetes = optional(object({\r\n        namespace = string\r\n        environmentNamespace = string\r\n      }))\r\n    })\r\n\r\n    azure = optional(object({\r\n      resourceGroup = object({\r\n        name = string\r\n        id = string\r\n      })\r\n      subscription = object({\r\n        subscriptionId = string\r\n        id = string\r\n      })\r\n    }))\r\n    \r\n    aws = optional(object({\r\n      region = string\r\n      account = string\r\n    }))\r\n  })"}}}
```

### Desired behavior

working tests pass. 

### Workaround

add .gitattributes 
```
*.tf text eol=lf
*.json text eol=lf
*.yaml text eol=lf
```
to the following folders:
* pkg/corperp/renderers/containers/testdata
* pkg/recipes/terraform/config/testdata
* pkg/recipes/terraform/testdata/.terraform/modules

### rad Version

v0.30.0-rc3-2-g2e4d2ed

### Operating system

Windows 11 with WSL 2 and Docker Desktop running the Devcontainer from the solution.

### Additional context

_No response_

### Would you like to support us?

- [X] Yes, I would like to support you

[AB#11104](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/11104)

### PR DESCRIPTION
**Pull request by jhandel**
# Description

Updated clivalidation.RequireResourceType so that it can take either fully qualified names or short names.. and if a short name is used and there are duplicate short names display the list of matching short names.

I also added a .gitattributes to the following folders so that their test files would maintain lf line ends instead of crlf line ends when cloned or pulled into a windows filesystem.

* pkg/corperp/renderers/containers/testdata
* pkg/recipes/terraform/config/testdata
* pkg/recipes/terraform/testdata/.terraform/modules

## Type of change

This pull request fixes a bug in Radius and has an approved issue 

Fixes: #7070

This pull request fixes a bug in Radius but its not an approved issue

Fixes:  #7121

### CODE REVIEW

File: pkg/cli/clivalidation.go [hunk 1]
```diff
@@ -270,13 +270,21 @@ func RequireResourceType(args []string) (string, error) {
...
 	resourceTypeName := args[0]
 	supportedTypes := []string{}
+	foundTypes := []string{}
 	for _, resourceType := range clients.ResourceTypesList {
 		supportedType := strings.Split(resourceType, "/")[1]
 		supportedTypes = append(supportedTypes, supportedType)
-		if strings.EqualFold(supportedType, resourceTypeName) {
-			return resourceType, nil
+		//check to see if the resource type is the correct short or long name.
+		if strings.EqualFold(supportedType, resourceTypeName) || strings.EqualFold(resourceType, resourceTypeName) {
```
Comment (vinayada1) [inline pkg/cli/clivalidation.go:278]: Could you add a unit test for this change?

**Comment by vinayada1** [comment]
@jhandel  - Thanks for your contribution.

File: pkg/cli/clivalidation.go (hunk 1 above)
Comment (jhandel) [reply pkg/cli/clivalidation.go:278]: @vinayada1,  it doesn't look like any of these functions in clivalidation are being independantly unit tested.. Do you want me to test this in the resources module (its used by list, show and delete) or should we start a clivalidation_test.go file and start it with this test?

Comment (vinayada1) [reply pkg/cli/clivalidation.go:278]: You are right! It would be great if you could start with a clivalidation_test.go file. Thanks!

**Review by lakshmimsft — APPROVED** [review]

**Pull request merged**